tests/test_datasets.py

```diff
@@ -24,3 +24,215 @@ def test_datasets_response_structure(
+                        },
+                        "pipeline_version": {
+                            "type": "literal",
+                            "value": "7.3.2",
+                        },
+                    },
+                    {
+                        "dataset_uuid": {
+                            "type": "uri",
+                            "value": "http://neurobagel.org/vocab/test-002",
+                        },
+                        "image_modal": {
+                            "type": "uri",
+                            "value": "http://purl.org/nidash/nidm#FlowWeighted",
+                        },
+                    },
+                    {
+                        "dataset_uuid": {
+                            "type": "uri",
+                            "value": "http://neurobagel.org/vocab/test-003",
+                        },
+                        "image_modal": {
+                            "type": "uri",
+                            "value": "http://purl.org/nidash/nidm#T1Weighted",
+                        },
+                    },
+                    {
+                        "dataset_uuid": {
+                            "type": "uri",
+                            "value": "http://neurobagel.org/vocab/test-003",
+                        },
+                        "image_modal": {
+                            "type": "uri",
+                            "value": "http://purl.org/nidash/nidm#FlowWeighted",
+                        },
+                    },
+                ]
+            },
+        }
+
+    expected_image_modals_and_pipelines = {
+        "http://neurobagel.org/vocab/test-001": {
+            "image_modals": [
+                "http://purl.org/nidash/nidm#T1Weighted",
+                "http://purl.org/nidash/nidm#FlowWeighted",
+            ],
+            "available_pipelines": {
+                "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/fmriprep": [
+                    "23.2.0"
+                ],
+                "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/mriqc": [
+                    "22.0.6"
+                ],
+            },
+        },
+        "http://neurobagel.org/vocab/test-002": {
+            "image_modals": [
+                "http://purl.org/nidash/nidm#T1Weighted",
+                "http://purl.org/nidash/nidm#FlowWeighted",
+            ],
+            "available_pipelines": {
+                "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/freesurfer": [
+                    "6.0.1",
+                    "7.3.2",
+                ]
+            },
+        },
+        "http://neurobagel.org/vocab/test-003": {
+            "image_modals": [
+                "http://purl.org/nidash/nidm#T1Weighted",
+                "http://purl.org/nidash/nidm#FlowWeighted",
+            ],
+            "available_pipelines": {},
+        },
+    }
+
```

Comment:
I don't usually like comments to structure the test, but if your setup is this long, it might be helpful to add # Setup  # Act # Assert to help a reader figure out where they need to start paying attention